### PR TITLE
📝Add suggestion level

### PR DESCRIPTION
**Pull request by mingwli**
#3838

### CODE REVIEW

File: src/docfx/lib/log/Error.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ internal class Error
     {
         public static readonly IEqualityComparer<Error> Comparer = new EqualityComparer();
 
-        public ErrorLevel Level { get; private set; }
+        public ErrorLevel Level { get; set; }
```
Comment (yufeih) [inline src/docfx/lib/log/Error.cs]:
```suggestion
        public ErrorLevel Level { get; }
```

File: src/docfx/lib/log/ErrorLevel.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ internal enum ErrorLevel
     {
         Off,
         Info,
+        Suggestion,
```
Comment (yufeih) [inline src/docfx/lib/log/ErrorLevel.cs]: Also needs to update Report.cs to limit suggestion count

File: src/docfx/lib/log/Error.cs [hunk 3]
```diff
@@ -65,7 +65,6 @@ public string ToString(ErrorLevel level)
 
         public DocfxException ToException(Exception innerException = null)
         {
-            Level = ErrorLevel.Error;
```
Comment (live1206) [inline src/docfx/lib/log/Error.cs]: without public 'set', this line could not be removed

Comment (live1206) [inline src/docfx/lib/log/Error.cs]: should this be kept?

**Pull request merged**